### PR TITLE
Update the border colors in the Cart and Checkout blocks

### LINKED ISSUE #11354: Update the colors in the Cart and Checkout blocks
We need to update the colors used for borders in the Cart and Checkout blocks to create a cohesive experience with the order confirmation block template defaults.

- Gutenberg Gray-100 For all borders except form fields
- Gutenberg Gray-900 For form fields

Before merging the work we need a confirmation from @elizaan36 on the final result. 

screenshot-2023-10-18-at-9 31 20-pm-1
screenshot-2023-10-18-at-9 57 02-pm
screenshot-2023-10-18-at-10 04 33-pm

### PR DESCRIPTION
**Pull request by tarunvijwani**
## What

Fixes #11354

## Why

In this PR, we're updating the border colors in the Cart and Checkout blocks to match the order confirmation block template defaults and create a cohesive experience.

## Testing Instructions

1. Go to the Cart block page and confirm the border color is similar to #E7E7E7.
2. Open the shipping calculator and confirm the form fields have a border color similar to #1E1E1E. 
3. Go to the Checkout block page and confirm the border color is similar to #E7E7E7
4. Confirm the form fields have a border color similar to #1e1e1e. 
5. Test it with different themes different background colors, and confirm that the borders are visible. 

* [ ] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

## Screenshots or screencast

https://github.com/woocommerce/woocommerce-blocks/assets/11503784/85714563-e608-474d-8e7b-92fb4351813e

## WooCommerce Visibility

Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:

* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:

* [x] This PR has a UI change and has been cross-browser tested at different viewport sizes on both the frontend and in the editor.
* [x] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [x] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> Update the border colors in the Cart and Checkout blocks.

### CODE REVIEW

**Comment by tarunvijwani** [comment]
This PR is on hold. There were concerns about user experience with light border colors. 
Waiting for @elizaan36 to confirm the colors for borders. 

Internal discussion: p1698400336133709/1698387117.246639-slack-C8X6Q7XQU

**Comment by tarunvijwani** [comment]
We decided to continue with $universal-border-light color for borders.

**Comment by tarunvijwani** [comment]
Hi @elizaan36, could you take a moment to review the PR? I've included a screen recording for reference. Just wanted to let you know that I added `rgba(0, 0, 0, 0.882)` color for `Gutengerb 900`, which is the same as `#1e1e1e` color. I'd really appreciate your feedback on this.

**Review by elizaan36 — COMMENTED** [review]
Hi @tarunvijwani I'm trying to figure out why, but the border color in the implementation is too light and almost disappears on the screen. 

It looks like the correct value from the code, but when I compare it visually to the Figma it's much lighter. 

image

Why do you think this is?

Also for the $universal-border-dark variable, could we still use the RGB value for consistency (17, 17, 17, .80) ? Rather than 0/0/0.

**Comment by tarunvijwani** [comment]
> It looks like the correct value from the code, but when I compare it visually to the Figma it's much lighter.

@elizaan36 I think that's because the rgba value is not correct. We're using rgba(17, 17, 17, 0.115); which is #e7e7e7 on white. However, on [Figma](https://www.figma.com/file/Xcn9ZirCFwU451IQZxMAcb/Block-based-checkout?node-id=5245%3A75642&mode=dev) we're using #E0E0E0;  Would you be able to help me with the correct rgba value for it?

image

**Comment by elizaan36** [comment]
No problem, I'll screenshot what I have now in Figma. For the `$universal-border-light` variable, I'm using `#111111 - 13%`. Which translates to the rgba value (17,17,17,.13). I'm also seeing that the equivalent is #e0e0e0.

Screenshot 2023-10-31 at 2 24 02 PM

Sorry for the confusion there as those Gutenberg values slipped into the mocks by accident. Hopefully this nails it, and I'll spend some time cleaning up the Figma.

One last thing. I noticed that the borders between the section numbers aren't following the same color system. Can we update those borders to the `$universal-border-light` variable as well? It creates a really nice consistency overall on the checkout.

Screenshot 2023-10-31 at 2 26 31 PM

**Comment by tarunvijwani** [comment]
Thank you, @elizaan36, for the color code. 

>One last thing. I noticed that the borders between the section numbers aren't following the same color system. Can we update those borders to the $universal-border-light variable as well? It creates a really nice consistency overall on the checkout.

I noticed form borders had opacity that's why they were appearing more lighter than the actual color. I have now removed the opacity from them.

Here is the updated output: 




https://github.com/woocommerce/woocommerce-blocks/assets/11503784/d6533c13-cff0-44bd-be42-17715654975d

**Comment by elizaan36** [comment]
I think we overcorrected a bit considering the additional transparency you mentioned, so tweaking the color one more time to the lighter rgba (17, 17, 17, .12) or even rgba (17, 17, 17, .11) would work.

**Comment by tarunvijwani** [comment]
@elizaan36 as we discussed over slack, I have rolled back the $universal-border-light value to rgba(17, 17, 17, 0.115). 

Here is the updated video:

## Storefront theme

https://github.com/woocommerce/woocommerce-blocks/assets/11503784/3edf50fb-a1f3-4175-9b37-04530ca7e52f


## Tove theme

https://github.com/woocommerce/woocommerce-blocks/assets/11503784/5f33aa12-1d7a-47c6-94b2-4bf8a909ff85

**Comment by elizaan36** [comment]
The border color looks perfect now. Thanks, @tarunvijwani 🙏

File: assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss [hunk 1]
```diff
@@ -52,11 +52,12 @@ $border-width: 1px;
 			border-left: $border-width solid currentColor;
 			border-top: $border-width solid currentColor;
 			border-radius: $universal-border-radius 0 0 0;
+			border-color: $universal-border-light;
 			content: "";
 			display: block;
 			height: $universal-border-radius - $border-width;
 			margin-right: $gap-small;
-			opacity: 0.3;
+			opacity: 1;
```
Comment (mikejolley) [inline assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss]: Are these values needed? 1 should be default.

Comment (tarunvijwani) [reply assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss]: @mikejolley While testing the PR, there were some instances where opacity was overridden to 0.3 again. I added this manually to make sure it remains the same and also doesn't inherit the theme opacity styles.

Comment (mikejolley) [reply assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss]: I think you should investigate where its coming from and fix there. It could be a mixin like `with-translucent-border`.

**Comment by mikejolley** [comment]
Moving to 11.6.0

File: assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss (hunk 1 above)
Comment (tarunvijwani) [reply assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss]: @mikejolley Thank you for the suggestion! I have passed opacity argument in the `with-translucent-border` as it accepts opacity and assign it to the psuedo element: 

```css

@mixin with-translucent-border($border-width: 1px, $opacity: 0.3) {
	position: relative;

	&::after {
		border-style: solid;
		border-width: $border-width;
		bottom: 0;
		content: "";
		display: block;
		left: 0;
		opacity: $opacity;
		pointer-events: none;
		position: absolute;
		right: 0;
		top: 0;
	}
}
```

**Comment by tarunvijwani** [comment]
@mikejolley Thank you for the suggestions, I have replaced `with-translucent-border` with normal borders. Please take a moment to review the PR again. 🙂

**Review by nielslange — COMMENTED** [review]
Thanks for working on this, @tarunvijwani. I quickly checked the PR against the Twenty Twenty-Four theme and some borders are pretty hard to read. 

## Style: Default

<table>
<tr>
<td valign="top">Before:
<br><br>
Screenshot 2023-11-08 at 16 37 46
</td>
<td valign="top">After:
<br><br>
Screenshot 2023-11-08 at 16 41 44
</td>
</tr>
</table>

## Style: Ember

<table>
<tr>
<td valign="top">Before:
<br><br>
Screenshot 2023-11-08 at 16 46 40
</td>
<td valign="top">After:
<br><br>
Screenshot 2023-11-08 at 16 41 31
</td>
</tr>
</table>

## Style: Maelstrom

<table>
<tr>
<td valign="top">Before:
<br><br>
Screenshot 2023-11-08 at 16 38 34
</td>
<td valign="top">After:
<br><br>
Screenshot 2023-11-08 at 16 41 18
</td>
</tr>
</table>

**Comment by mikejolley** [comment]
@nielslange I don't think we should get too caught up in supporting every possible combination because we're trying to create a generic solution that works in most cases. There will clearly be borders that need to be set by the merchant when creating the blocks with global styles. 

Do you have thoughts on this @ralucaStan?

File: assets/css/abstracts/_colors.scss [hunk 2]
```diff
@@ -20,3 +20,4 @@ $image-placeholder-border-color: #f2f2f2;
 $universal-border: rgba(17, 17, 17, 0.3); // Used for form step borders.
 $universal-border-light: rgba(17, 17, 17, 0.115); // e7e7e7 on white.
 $universal-body-low-emphasis: rgba(17, 17, 17, 0.5); // Used for low emphasis text such as input labels.
+$universal-border-dark: rgba(17, 17, 17, 0.8); // #1e1e1e on white. Used for input borders.
```
Comment (mikejolley) [inline assets/css/abstracts/_colors.scss]: This is fine for now but we may need to revisit naming depending on if it makes sense to target elements (-input-border) vs shades (-border-dark).

File: assets/js/base/components/cart-checkout/order-summary/style.scss [hunk 3]
```diff
@@ -27,6 +27,8 @@
 		}
 
 		&:last-child {
+			border-width: 0;
```
Comment (mikejolley) [inline assets/js/base/components/cart-checkout/order-summary/style.scss]: Is this to reset the style above? I would prefer to see `border-bottom: 0` if so, just resetting what you added before.

File: assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss [hunk 4]
```diff
@@ -81,12 +81,12 @@ $border-width: 1px;
 	}
 
 	.wc-block-components-express-payment__content {
-		@include with-translucent-border(0 $border-width $border-width);
+		border-style: solid;
+		border-width: 0 $border-width $border-width;
+		bottom: 0;
+		border-color: $universal-border-light;
```
Comment (mikejolley) [inline assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss]: I'm not sure what you prefer, but if it is more succinct this could be written with:

```suggestion
		border: $border-width solid $universal-border-light;
		border-top: 0;
		bottom: 0;
```

File: assets/js/blocks/cart-checkout-shared/payment-methods/style.scss [hunk 5]
```diff
@@ -198,7 +198,11 @@
 
 	.wc-block-components-radio-control-accordion-option,
 	.wc-block-components-radio-control__option {
-		@include with-translucent-border(1px 1px 0 1px);
+		border: 1px solid $universal-border-light;
```
Comment (mikejolley) [inline assets/js/blocks/cart-checkout-shared/payment-methods/style.scss]: This one had no bottom border previously—was this change intentional?

**Review by mikejolley — COMMENTED** [review]
There are a few more instances of `$input-border-gray`. Should we replace those and remove the style rule so its not used going forward?

**Comment by nielslange** [comment]
> @nielslange I don't think we should get too caught up in supporting every possible combination because we're trying to create a generic solution that works in most cases. [...]

It's not my intension to find a solution for each case. I simply wanted to point out that the current solution has a very low color contrast ratio. IMO, these colors are far from a generic solution as some borders are very hard to see that way.

**Comment by nielslange** [comment]
> @nielslange I don't think we should get too caught up in supporting every possible combination because we're trying to create a generic solution that works in most cases. [...]

@mikejolley It's not my intension to find a solution for each case. I simply wanted to point out that the current solution has a very low color contrast ratio. IMO, these colors are far from a generic solution as some borders are very hard to see that way. A generic solution, should work for the majority of users, independent of certain conditions, such as eye disorders. 

https://wordpress.org/about/accessibility/ states:

> WordPress aims to make the WordPress Admin and bundled themes fully WCAG 2.0 AA compliant where possible.

Regarding the color contrast, https://make.wordpress.org/accessibility/handbook/design/use-of-color/#color-contrast states:

> - 4.5:1 for normal text
> - 3:1 for [[large text (24px equivalent or 19px equivalent and bold)](http://mcdpartners.com/lab/meeting-wcag-color-contrast-guideline/)](http://mcdpartners.com/lab/meeting-wcag-color-contrast-guideline/)

Using the default style of the Twenty Twenty-Four theme, without applying any modifications, results in [a color contrast ratio of 1.27:1](https://coolors.co/contrast-checker/dfdfdf-f9f9f9).

Screenshot 2023-11-08 at 20 54 07

My primary intension is to prevent excluding folks with vision impairment from using WooCommerce Blocks due to accessibility reasons.

**Comment by mikejolley** [comment]
@nielslange that resource refers to text though doesn't it. It's my understanding that this PR and the designs in Figma only adjust border colors.

**Comment by nielslange** [comment]
> @nielslange that resource refers to text though doesn't it. It's my understanding that this PR and the designs in Figma only adjust border colors.

It's true that this PR only affects borders. That said, I consider borders as visual elements that should be visible. The moment their contrast ratio is too low, that's the moment when these visual elements start to become invisible for visual impaired users. I do not want to block this PR, though. I only want to raise my concerns that by applying these border colors we will exclude certain people from seeing them.

**Comment by ralucaStan** [comment]
@nielslange what borders are failing the a11y check? If it is only the decoration ones used for visual separation those can be ignored IMO. They only have a decorative purpose

**Comment by nielslange** [comment]
As I said, I don't want to block this PR. I only want to state that this way forward will affect visually impaired people, @ralucaStan .

**Comment by ralucaStan** [comment]
@nielslange I understand and appreciate the intention. My question still holds. Which borders are failing the a11y check? Because if the form elements borders are failing that might lead to some people not being able to fill in the form. The decoration borders don't have an impact in the checkout process.

**Comment by nielslange** [comment]
> @nielslange I understand and appreciate the intention. My question still holds. Which borders are failing the a11y check? Because if the form elements borders are failing that might lead to some people not being able to fill in the form. The decoration borders don't have an impact in the checkout process.

Currently, only decorative borders fail the accessibility check. However, I do not understand why we have decorative borders, knowing that they will not be visible to everyone. IMO, we should either have borders that are visible to all users, independent of visual impairment, or no borders at all. Having borders, that are only visible to non-visually impaired users, sounds very exclusive to me. Personally, I'm not visually impaired, yet I already have a problem in seeing the decorative borders on the following screenshot, which shows the Twenty Twenty-Four theme without any adjustments:

Screenshot 2023-11-08 at 22 47 02

As mentioned before, it's not my intension to block this PR in any way. If you feel that we should do that, then please go ahead. Personally, I doubt that this should be the way forward, as we do exclude certain users from seeing specific elements, even if we only consider them as decorative. My primary intension is that the UI of WooCommerce Blocks is as inclusive as possible. IMO, this PR is rather exclusive than inclusive.

**Comment by mikejolley** [comment]
@nielslange I understand the point you're trying to put across but saying stuff like this you're basically saying that if we merge this we don't care about users with impaired vision? This isn't really true. 

> If you feel that we should do that, then please go ahead. Personally, I doubt that this should be the way forward, as we do exclude certain users from seeing specific elements, even if we only consider them as decorative.

The accessibility docs apply to text. @ralucaStan was correct to also point out that important non-decorative elements should also have sufficient contrast (input borders) which you've confirmed is the case. It looks like the designs in Figma took important/decorative borders into account already.

> However, I do not understand why we have decorative borders, knowing that they will not be visible to everyone. IMO, we should either have borders that are visible to all users, independent of visual impairment, or no borders at all. 

By this logic everything would be plain text, otherwise we'd exclude users of screen readers too :)

I'd vote to proceed with the change. @tarunvijwani maybe that would be good to document alongside the definitions of the universal colors; ensure that the low contrast colors are marked as used for decorative elements only so its not applied to inputs or text by accident in the future.

**Comment by ralucaStan** [comment]
> I'd vote to proceed with the change. @tarunvijwani maybe that would be good to document alongside the definitions of the universal colors; ensure that the low contrast colors are marked as used for decorative elements only so its not applied to inputs or text by accident in the future.

Sounds good to me. Thank you all for your input. I appreciate the care. Should we see any impact of this PR, as always, we'll be quick to act on it.

**Comment by tarunvijwani** [comment]
@nielslange, thank you for taking the time to comment. 
I appreciate your concerns about the border colors, and I have a potential solution that we can consider. We can add an option for the merchants to change the color of the decorative borders. This way, we'll be aligned with FSE and provide more control to the merchants in styling their sites. Currently, they don't have this option, and I agree that it's not a good user experience. However, I understand that adding this feature would require extra effort. We can consider opening a new ticket for it and work on it in a later cycle. 
@ralucaStan @mikejolley Please let me know your thoughts on this.

**Comment by tarunvijwani** [comment]
>@tarunvijwani maybe that would be good to document alongside the definitions of the universal colors; ensure that the low contrast colors are marked as used for decorative elements only so its not applied to inputs or text by accident in the future.

@mikejolley That's a great suggestion. I'll document the definitions. 👍

**Comment by ralucaStan** [comment]
@tarunvijwani does not hurt to open a new issue, but make it a low priority. Thank you

File: assets/js/blocks/cart-checkout-shared/payment-methods/style.scss (hunk 5 above)
Comment (tarunvijwani) [reply assets/js/blocks/cart-checkout-shared/payment-methods/style.scss]: Yes, this change was intentional. Previously, the borders were added on pseudo elements.

**Comment by tarunvijwani** [comment]
@mikejolley I have made all the suggested changes. Please take a moment to review the PR again. Thank you so much!

**Review by mikejolley — CHANGES_REQUESTED** [review]
Code changes look good and I reviewed the cart/checkout flow and it seems to be working as intended. I noticed one issue that needs resolving. There are gaps on the top borders in cart. See:

Screenshot 2023-11-14 at 10 59 23

**Comment by tarunvijwani** [comment]
@mikejolley Fixed border gap in https://github.com/woocommerce/woocommerce-blocks/pull/11474/commits/f1603e74cec083fafd1b152e5053f2524c82cc0e 👍

**Comment by mikejolley** [comment]
@tarunvijwani now there is no top border at all:
Screenshot 2023-11-15 at 14 18 02

**Comment by ralucaStan** [comment]
@tarunvijwani this PR is still using the old PR template so I wanted to make sure you tested these UI changes on Firefox, Safari as well and on mobile

**Comment by mikejolley** [comment]
@tarunvijwani I checked out the latest changes and the top border has now vanished in editor and frontend. This needs further testing?

Screenshot 2023-11-15 at 14 18 02

**Comment by tarunvijwani** [comment]
> @tarunvijwani this PR is still using the old PR template so I wanted to make sure you tested these UI changes on Firefox, Safari as well and on mobile

Yes, I have tested it on different browsers and screen sizes. I have updated the template of the PR as well. Thank you, @ralucaStan! 

>@tarunvijwani I checked out the latest changes and the top border has now vanished in editor and frontend. This needs further testing?

@mikejolley I have fixed the borders now. Here is the screen recording of the final version(Updated the video the PR description): 

https://github.com/woocommerce/woocommerce-blocks/assets/11503784/85714563-e608-474d-8e7b-92fb4351813e

**Comment by mikejolley** [comment]
Checkboxes are using the wrong color @tarunvijwani 

Screenshot 2023-11-16 at 12 16 08

**Review by mikejolley — CHANGES_REQUESTED** [review]
Checkboxes are using the wrong color @tarunvijwani 

Screenshot 2023-11-16 at 12 16 08

I would review Figma before marking this ready to review again to ensure its matching.

File: assets/js/blocks/checkout/inner-blocks/checkout-pickup-options-block/style.scss [hunk 6]
```diff
@@ -50,7 +50,7 @@
 	.wc-block-components-radio-control__description,
 	.wc-block-components-radio-control__secondary-description {
 		width: 100%;
-		text-align: left;
+		text-align: left !important; // To force align left on the description for the local pickup options.
```
Comment (tarunvijwani) [inline assets/js/blocks/checkout/inner-blocks/checkout-pickup-options-block/style.scss]: On trunk, the local pick-up description was aligned right.  Added `!important` to force align left.

| Before | After |
|---|---|
|image|image|

**Comment by tarunvijwani** [comment]
@elizaan36 Couple of questions related to latest Figma design: Xcn9ZirCFwU451IQZxMAcb-fi-4749%3A86624:

## 1. Checkout Block: Shipping method background color:

image

In the design, there are a couple of mockups, and a few of them do not have any background color for the shipping method that is not selected. Can you confirm that there will be no color for the unselected shipping method?


## 2. Checkout Block: Accent color focus shadow:

Should we add accent color focus shadow for the selected text input?

image


## 3. Cart Block:  Quantity Selector
image
The border color for the quantity is not matching with the overall border color. Please confirm the border color for quantity. I believe it should match with all the borders to keep it consistent.
Example: 

image

**Comment by elizaan36** [comment]
> In the design, there are a couple of mockups, and a few of them do not have any background color for the shipping method that is not selected. Can you confirm that there will be no color for the unselected shipping method?

@tarunvijwani In `pdFofs-1dB#comment-1975` I shared details on the interaction of the shipping method section in a video. The background appears on hover and active states only, there's no background color for the unselected shipping method.

> Should we add accent color focus shadow for the selected text input?

Yes! We can include this as a future enhancement in another issue, I think it would improve accessibility and personality but needs to be tested.

> Please confirm the border color for quantity. I believe it should match with all the borders to keep it consistent.
Example:

Good call out and your example looks great. I updated the Figma to show the $universal-border-light variable for the Cart and quantity picker outline. 
image

File: assets/js/blocks/checkout/inner-blocks/checkout-pickup-options-block/style.scss (hunk 6 above)
Comment (tarunvijwani) [reply assets/js/blocks/checkout/inner-blocks/checkout-pickup-options-block/style.scss]: Reverted the change as it got fixed in trunk

**Comment by ralucaStan** [comment]
@tarunvijwani if this PR is still in progress then let's make it a draft until it's good for review again.

**Comment by tarunvijwani** [comment]
@elizaan36 As discussed, I'm converting the PR to draft till we finalize the design. 

Here is the link to the JN sites for your reference: 
This PR:  https://bare-spider.jurassic.ninja/
WC Blocks 11.6.1 version: https://serious-scarlet.jurassic.ninja/

**Comment by elizaan36** [comment]
Thanks @tarunvijwani All looks good, a couple notes.

I discussed this with the design team and we need to define a medium gray for the resting state of both radio buttons and checkboxes because the light gray looks disabled and the strong border color is too noisy for these selectors.

So overall we have three grays:
- $universal-border-light:rgba (17, 17, 17, .11); For section borders
- $universal-border-medium:rgba (25, 23, 17, .3); For radio and checkbox resting state
- $universal-border-strong: rgba(17, 17, 17, 0.8); For input borders.

checkout clean up

Also, in the prototype it looks like the radio buttons aren't horizontally aligned. The shipping and payment method selections should appear in the same vertical row.

image

Figma: Xcn9ZirCFwU451IQZxMAcb-fi

**Comment by tarunvijwani** [comment]
Thank you, @elizaan36, for sharing the final design. I have incorporated all the changes in this PR and created a JN site for you to have a look: https://apparent-orca.jurassic.ninja/. Please take a moment to review the site and let me know if any additional changes are required. 

For `$universal-background` color, I kept it `rgba(17, 17, 17, 0.02)` as `rgba(17, 17, 17, 0.2)` was appearing too dark.

Here is the screenshot for `rgba(17, 17, 17, 0.2)`:


image

**Comment by elizaan36** [comment]
Looks great! 🙏
You were right about the color. I mistakenly wrote .2 but 2% = .02.

**Comment by tarunvijwani** [comment]
@elizaan36 Thank you for the confirmation. 
@nielslange @mikejolley The PR is ready for review. 🙂

**Review by nielslange — COMMENTED** [review]
Thanks for working on this, @tarunvijwani. I've been reviewing the contrast aspects in line with our discussions and the WCAG guidelines. While the [Text Contrast guidelines](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) focus on text elements, the [Non-Text Contrast guidelines](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html) also emphasize the importance of sufficient contrast for UI components, including checkboxes and radio buttons.

I noticed that in the default styles of the Twenty Twenty-Four theme, checkboxes and unselected radio buttons have a contrast ratio of 1.86:1, which is below the recommended 3:1 ratio. This could impact users with visual impairments. These are the UI elements I'm referring to:

Screenshot 2023-12-06 at 16 10 41

Would it be possible to adjust these elements to align with the 3:1 contrast ratio as suggested by the guidelines? This enhancement could significantly improve accessibility for all users.

cc: @elizaan36

**Comment by tarunvijwani** [comment]
Increasing the contrast of the border color for the checkbox and radio buttons could lead to an inconsistent look. However, we should consider the WCAG guidelines. I will wait for @elizaan36 to provide her suggestions.

**Comment by elizaan36** [comment]
We can bump the $universal-border-medium to `rgba (25, 23, 17, .48)`

The increase in opacity passes in my contrast checker, let me know if that passes on your end.

**Comment by nielslange** [comment]
> We can bump the $universal-border-medium to `rgba (25, 23, 17, .48)`
> 
> The increase in opacity passes in my contrast checker, let me know if that passes on your end.

Thanks for considering this change, @elizaan36. I want to clarify that my intention isn't to block this PR. I'm simply very passionate about a11y and strive to ensure that all UI elements are equally accessible.

**Comment by tarunvijwani** [comment]
Thank you, @nielslange and @elizaan36! I have updated the colors, here is how it looks like on the Twenty Twenty Four theme:

image

File: assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss [hunk 7]
```diff
@@ -81,12 +81,11 @@ $border-width: 1px;
 	}
 
 	.wc-block-components-express-payment__content {
-		@include with-translucent-border(0 $border-width $border-width);
+		border: $border-width solid $universal-border-light;
+		border-top: 0;
+		bottom: 0;
```
Comment (nielslange) [inline assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss]: `bottom: 0;` does not seem to have any effect here:

Screenshot 2023-12-07 at 15 28 58

**Review by nielslange — COMMENTED** [review]
Thanks for working on this, @tarunvijwani. Overall, the PR looks good. I just noticed one definition that doesn't seem to do anything.

File: assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss (hunk 7 above)
Comment (tarunvijwani) [reply assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss]: @nielslange Nice catch! Thank you, I have removed it.

Comment (nielslange) [reply assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/style.scss]: Thanks for adjusting this part.

**Review by nielslange — APPROVED** [review]
Looking great now, @tarunvijwani. Let's ⛴️ this change. 🙌

**Review by elizaan36 — COMMENTED** [review]
LGTM! Thanks!
